numpy/core/src/multiarray/compiled_base.c

```diff
@@ -80,6 +80,374 @@ minmax(const npy_intp *data, npy_intp data_len, npy_intp *mn, npy_intp *mx)
+                 */
+                if (ix == bins || (ix != 0 && tx < e[ix])) {
+                    ix--;
+                }
+                else if ((ix + 1) != bins && tx >= e[ix + 1]) {
+                    ix++;
+                }
+                ihist[ix]++;
+            }
+        }
+        NPY_END_ALLOW_THREADS;
+    }
+    else {
+        npy_intp i;
+        double *x, *w, *dhist;
+
+        arr_w = (PyArrayObject *)PyArray_ContiguousFromAny(weights, NPY_DOUBLE, 1, 1);
+        if (arr_w == NULL) {
+            goto fail;
+        }
+        if (PyArray_SIZE(arr_w) != len) {
+            PyErr_SetString(PyExc_ValueError,
+                            "Weights must be same length as array");
+            goto fail;
+        }
+
+        hist = (PyArrayObject *)PyArray_ZEROS(1, &bins, NPY_DOUBLE, 0);
+        if (hist == NULL) {
+            goto fail;
+        }
+
+        x = (double *)PyArray_DATA(arr_x);
+        w = (double *)PyArray_DATA(arr_w);
+        dhist = (double *)PyArray_DATA(hist);
+
+        NPY_BEGIN_ALLOW_THREADS;
+        for(i = 0; i < len; i++) {
+            double tx = x[i];
+            if ((min <= tx) && (tx <= max)) {
+                npy_intp ix = (tx - min) * norm;
+                if (ix == bins || (ix != 0 && tx < e[ix])) {
+                    ix--;
+                }
+                else if ((ix + 1) != bins && tx >= e[ix + 1]) {
+                    ix++;
+                }
+                dhist[ix] += w[i];
+            }
+        }
+        NPY_END_ALLOW_THREADS;
+
+        Py_DECREF(arr_w);
+    }
+
+    Py_DECREF(arr_edges);
+    Py_DECREF(arr_x);
+    return (PyObject *)hist;
+
+fail:
+    Py_XDECREF(hist);
+    Py_XDECREF(arr_w);
+    Py_XDECREF(arr_edges);
+    Py_XDECREF(arr_x);
+    return NULL;
+}
+
+NPY_NO_EXPORT PyObject *
+arr_histogramdd_uniform(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
+{
+    PyObject *obj_x, *obj_bins, *obj_edges, *weights = Py_None;
+    PyArrayObject *arr_x = NULL, *arr_bins = NULL, *arr_edges = NULL,
+                  *arr_w = NULL, *hist = NULL;
+    npy_intp len, ndim, *bins;
+    double min[NPY_MAXDIMS], max[NPY_MAXDIMS], norm[NPY_MAXDIMS];
+
+    static char *kwlist[] = {"x", "nbins", "edges", "weights", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OOO|O:_histogramdd_uniform",
+            kwlist, &obj_x, &obj_bins, &obj_edges, &weights)) {
+        goto fail;
+    }
+
+    /*
+     * Get data (x) which is of shape (len, ndim)
+     * where len is the number of samples
+     * and ndim is the number of dimensions
+     */
+    arr_x = (PyArrayObject *)PyArray_ContiguousFromAny(obj_x, NPY_DOUBLE, 2, 2);
+    if (arr_x == NULL) {
+        goto fail;
+    }
+    len = PyArray_DIM(arr_x, 0);
+    ndim = PyArray_DIM(arr_x, 1);
+
+    /*
+     * Get the number of bins for each array since
+     * the edges are passed in as a flat array
+     */
+    arr_bins = (PyArrayObject *)PyArray_ContiguousFromAny(obj_bins, NPY_INTP, 1, 1);
```

Comment:
Does this accept and round floating point values, when it ought to be erroring?

Reply:
So my first attempt to add type checking on `obj_bins` failed and I haven't yet figured out a way to do so. Any help would be much appreciated.

As it stands, this array is created in `numpy/lib/function_base.py` and so will always be integers. If it does get floats and rounds down it will be caught in the check of the total number bins later in the function.